Formula/s2geometry.rb

```diff
@@ -19,69 +18,108 @@ class S2geometry < Formula
   end
 
   depends_on "cmake" => :build
-  depends_on "glog" => :build
-  depends_on "googletest" => :build
```

Comment:
Upstream doesn't support our `googletest` formula. They require building with the source code and mention old v1.8.0. This also causes install to package conflicting gtest libraries/headers. `googletest` usage is listed as "to build tests and example programs, optional", so decided to just drop it.